server/src/core/server/graph/resolvers/User.ts

```diff
@@ -106,4 +106,19 @@ export const User: GQLUserTypeResolver<user.User> = {
     const tenDaysAgo = new Date(Date.now() - 10 * 24 * 60 * 60 * 1000);
     return lastFeaturedDate && lastFeaturedDate >= tenDaysAgo;
   },
+  badges: ({ badges, secretBadges }) => {
+    if (!badges && !secretBadges) {
+      return undefined;
```

Comment:
I did this to retain the original behaviour of the `badges` possibly being `undefined` and thus making sure we aren't messing things up by returning an empty array when something might expect `undefined` instead. I doubt there is any issue with returning an empty array, but why mess with an edge case that might exist?